deployment/helm/templates/custom-discovery.yaml

```diff
@@ -0,0 +1,93 @@
+{{- if .Values.customDiscovery.enabled }}
+apiVersion: {{ printf "%s/%s" .Values.crds.group .Values.crds.version }}
+kind: Configuration
+metadata:
+  name: {{ .Values.customDiscovery.name }}
+spec:
+  protocol: 
+    name: {{ .Values.customDiscovery.protocolName }}
+    discoveryDetails:
+      {{- range $key, $val := .Values.customDiscovery.discoveryDetails }}
+      {{ $key }}: {{ $val }}
+      {{- end }}
+  {{- if .Values.customDiscovery.brokerPod.image }}
+  brokerPodSpec:
+    containers:
+    - name: {{ .Values.customDiscovery.name }}-broker
+      image: {{ .Values.customDiscovery.brokerPod.image | quote }}
+      {{- with .Values.customDiscovery.pullPolicy }}
+      imagePullPolicy: {{ . }}
+      {{- end }}
+      resources:
+        limits:
+          {{`"{{PLACEHOLDER}}"`}} : "1"
+    {{- with .Values.imagePullSecrets }}
+    imagePullSecrets:
+      {{- toYaml . | nindent 6 }}
+    {{- end }}
+  {{- end }}
+  {{- if .Values.customDiscovery.brokerPod.image }}
+  {{- if .Values.customDiscovery.createInstanceServices }}
+  instanceServiceSpec:
+    type: {{ .Values.customDiscovery.instanceService.type }}
+    ports:
+    - name: {{ .Values.customDiscovery.instanceService.name }}
+      port: {{ .Values.customDiscovery.instanceService.port }}
+      protocol: {{ .Values.customDiscovery.instanceService.protocol }}
+      targetPort: {{ .Values.customDiscovery.instanceService.targetPort }}
+  {{- end }}
+  {{- if .Values.customDiscovery.createConfigurationService }}
+  configurationServiceSpec:
+    type: {{ .Values.customDiscovery.configurationService.type }}
+    ports:
+    - name: {{ .Values.customDiscovery.configurationService.name }}
+      port: {{ .Values.customDiscovery.configurationService.port }}
+      protocol: {{ .Values.customDiscovery.configurationService.protocol }}
+      targetPort: {{ .Values.customDiscovery.configurationService.port }}
+  {{- end }}
+  {{- end }}
+  {{- if .Values.customDiscovery.properties }}
+  properties:
+  {{- toYaml . | nindent 4 }}
+  {{- else }}
+  properties: {}
+  {{- end }}
+  capacity: {{ .Values.customDiscovery.capacity }}
+{{- end }}
+---
+{{- if .Values.customDiscovery.discovery.enabled }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: akri-custom-discovery-discovery-daemonset
+spec:
+  selector:
+    matchLabels:
+      name: akri-custom-discovery-discovery
```

Comment:
seems like some of these labels might want to be built using .Values.customDiscovery.name in case someone specifies more than one custom discovery.

Reply:
Good point. If they wanted to deploy the same template twice, they would have to do two installs right? Such as:
```
helm install http-config akri-helm-charts/akri \
  --set customDiscovery.discovery.enabled=true  \
  --set customDiscovery.name=akri-http \
  --set customDiscovery.discovery.image.repository=IMAGE
 ```

Reply:
i'm not sure that 2 installs would work ... each node can only have one agent.  this is interesting.

Reply:
i think we'd need 1 helm install, so that there is only one agent.

Reply:
The could set `agent.enabled=false` and `controller.enabled=false` in both of them in which case the installs are being used solely to add Configurations.

Reply:
sounds neat ... we should try that to verify that it works.

Reply:
Just tested out the following flow (1) apply Akri core components (agent and controller), (2) apply 2 different custom configurations, (3) apply a custom discovery handler. These could have been reordered in any way:

JUST INSTALL AKRI
`helm install akri akri-0.3.0.tgz`

JUST APPLY CONFIGURATIONS
```
helm install zeroconf-config akri-0.3.0.tgz \
 --set controller.enabled=false \
 --set agent.enabled=false \
 --set rbac.enabled=false \
 --set customDiscovery.enabled=true  \
 --set customDiscovery.protocolName=zeroconf \
 --set customDiscovery.name=akri-zeroconf 
```
```
helm install http-config akri-0.3.0.tgz \
 --set controller.enabled=false \
 --set agent.enabled=false \
 --set rbac.enabled=false \
 --set customDiscovery.enabled=true  \
 --set customDiscovery.protocolName=http \
 --set customDiscovery.name=akri-http 
```
JUST APPLY HTTP DISCOVERY DAEMONSET
```
helm install http-discovery akri-0.3.0.tgz \
 --set controller.enabled=false \
 --set agent.enabled=false \
 --set rbac.enabled=false \
 --set customDiscovery.discovery.enabled=true  \
 --set customDiscovery.name=akri-http-2 \
 --set customDiscovery.discovery.image.repository="nginx:latest"
 ```
 JUST DELETE ZEROCONF CONFIGURATION
 `helm delete zeroconf-config`
  JUST DELETE HTTP CONFIGURATION
 `helm delete http-config`
 JUST DELETE HTTP DISCOVERY
 `helm delete http-discovery`
 JUST DELETE AKRI
 `helm delete akri`
 
 While this all works, it does illustrate something confusing: in order to apply a Configuration, you set `customDiscovery.enabled=true` or `udev.enabled=true`. This made sense when we only were applying Configurations for protocols. **We might want to change this to `udev.configuration.enabled=true`;** however, i didnt include this in this PR because that would change the user interface and a lot of docs. What do you think about putting this change in the next PR that includes documentation on use discovery pods and agent-slim?